pkgs/misc/cups/drivers/foomatic-db-ppds/foomatic-ppdfiles.pl

```diff
@@ -0,0 +1,82 @@
+#! /usr/bin/env perl
+
+
+# The package `foomatic-db-engine` contains a
+# Perl script `foomatic-ppdfile` that can also
+# be used to extract ppd files from the database.
+# However, that script only produces one ppd file per call,
+# but it needs to process the complete database to do so.
+# To accelerate the process (about two orders of magnitude),
```

Comment:
Have you considered contributing this script upstream? With or without this script part of `foomatic-db-engine`, I envision that `foomatic-db-ppds` could be removed, and you'll run that script in `foomatic-db{,-nonfree}` (perhaps conditionally) - making these packages include the ppd files from the databases they came with.

Reply:
Thanks for bringing up this idea.  The `foomatic-db-engine` maintainer [answered promptly](https://github.com/OpenPrinting/foomatic-db-engine/pull/8#issuecomment-1296183602) and brought another command to my attention:  I didn't spot `foomatic-compiledb` earlier, but it is doing exactly what I tried to accomplish with my tiny script.  Up to the subdirectory structure, the result is the same, so we don't have to patch `foomatic-db-engine` at all.

Note: I also upgraded the `foomatic-db-engine` from 4.0 to 4.1 while preparing the patch.  With this update, the final `foomatic-db-ppds` package "lost" about 1.000 PPD files, but apparently these files were duplicates of files in `foomatic-db`, so we don't really loose anything.

Regarding the ppd export directly in the database packages: Inspired by your idea, I restructured everything to make this work, but alas, `foomatic-db-nonfree` isn't a "database" by itself, but just a kind of extension to the free `foomatic-db`.  The free database would install and export everything alright, but the nonfree version would need to be combined with the free one after installation, only to generate the free ppd files again.  The result of this restructuring would produce working packages, but the build script of the nonfree version would work very differently (and more complicated) compared to the free version, leading to a quite awkward structure overall.  Finally, I came to the conclusion that the current design is more understandable and maintainable, so I suggest to keep it.

Reply:
The current packages' structure is great, I'm glad you switched to `foomatic-compiledb`. Thanks for the effort!